### PR TITLE
improvements to log chart

### PR DESCRIPTION
**Pull request by jakubczakon**
* moved log_chart from `neptunecontrib.logging` to `neptunecontrib.api` (and added deprecation warning)
* moved pickle_and_log_artifact helper to `neptunecontrib.api` (and added deprecation warning)
* added support for bokeh and altair
* added nicer examples and links to log chart docstrings
* added log_table for pandas dataframes

### CODE REVIEW

File: neptunecontrib/api/chart.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import neptune
+
+
+def log_chart(name, chart, experiment=None):
+    """Logs charts from matplotlib, plotly to neptune.
```
Comment (pitercl) [inline neptunecontrib/api/chart.py:21]: Maybe add bokeh and altair here as well? Or rephrase somehow?

File: neptunecontrib/monitoring/utils.py [hunk 2]
```diff
@@ -65,74 +65,21 @@ def axes2fig(axes, fig=None):
...
-            from neptunecontrib.monitoring.utils import send_figure
-
-            with neptune.create_experiment():
-                fig, ax = plt.subplots()
-                sns.heatmap(table,ax=ax)
-                send_figure(fig)
-
+    message = """neptunecontrib.monitoring.utils send_figure functionality is now available in neptune-client.
+    You should simply use neptune.log_image('channel_name', fig) where you used send_figure('channel_name', fig) before.
+    send_figure will be depracated in future releases.
```
Comment (pitercl) [inline neptunecontrib/monitoring/utils.py:70]: will be removed

File: neptunecontrib/monitoring/utils.py [hunk 3]
```diff
@@ -65,74 +65,21 @@ def axes2fig(axes, fig=None):
...
-
-            from sklearn.ensemble import RandomForestClassifier
-            from neptunecontrib.monitoring.utils import pickle_and_send_artifact
-
-            with neptune.create_experiment():
-                rf = RandomForestClassifier()
-                pickle_and_send_artifact(rf, 'rf')
+    message = """neptunecontrib.monitoring.utils pickle_and_send_artifact was moved to neptunecontrib.api
+    and renamed to pickle_and_log_artifact. You should use ``from neptunecontrib.api import pickle_and_log_artifact``
+    neptunecontrib.logging.log_chart will be deprecated in future releases.
```
Comment (pitercl) [inline neptunecontrib/monitoring/utils.py:81]: will be removed

File: neptunecontrib/logging/chart.py [hunk 4]
```diff
@@ -13,102 +13,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import warnings
 
-import neptune
+message = """neptunecontrib.logging.chart was moved to neptunecontrib.api.
+You should use ``from neptunecontrib.api import log_chart`` 
+neptunecontrib.logging.log_chart will be deprecated in future releases.
```
Comment (pitercl) [inline neptunecontrib/logging/chart.py:20]: will be removed

File: neptunecontrib/monitoring/xgboost_monitor.py [hunk 5]
```diff
@@ -13,226 +13,11 @@
...
 # limitations under the License.
 #
-import os
-import tempfile
+import warnings
 
-import neptune
-import xgboost as xgb
+message = """neptunecontrib.monitoring.xgboost_monitor was moved to neptunecontrib.monitoring.xgboost
+neptunecontrib.monitoring.xgboost_monitor will be deprecated in future releases.
```
Comment (pitercl) [inline neptunecontrib/monitoring/xgboost_monitor.py:19]: will be removed

File: neptunecontrib/versioning/__init__.py [hunk 6]
```diff
@@ -13,3 +13,5 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+
+from neptunecontrib.versioning.data import *
```
Comment (pitercl) [inline neptunecontrib/versioning/__init__.py:17]: Please import only what we actually want to expose. Alternatively use `__all__`.

File: neptunecontrib/viz/__init__.py [hunk 7]
```diff
@@ -13,3 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+
+from neptunecontrib.viz.experiments import *
```
Comment (pitercl) [inline neptunecontrib/viz/__init__.py:17]: Please import only what we actually want to expose. Alternatively use `__all__`.

File: neptunecontrib/hpo/__init__.py [hunk 8]
```diff
@@ -13,3 +13,5 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+
+from neptunecontrib.hpo.utils import *
```
Comment (pitercl) [inline neptunecontrib/hpo/__init__.py:17]: Please import only what we actually want to expose. Alternatively use `__all__`.

File: neptunecontrib/api/__init__.py [hunk 9]
```diff
@@ -13,3 +13,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+
+from neptunecontrib.api.chart import *
```
[the remark this answers is not in the document]
Comment (jakubczakon) [reply neptunecontrib/api/__init__.py:17]: IMHO It makes it explicit but really verbose for the users -> they want to use api helpers and don't care where they are or how we structured the lib.

This is a design decision, of course, but I feel that the usage feel of neptune-contrib due to those explicit imports was verbose and clunky. And so I wanted to change that.
I am ok with leaving it explicit just my feel is against it.

File: neptunecontrib/hpo/__init__.py (hunk 8 above)
Comment (jakubczakon) [reply neptunecontrib/hpo/__init__.py]: Done

File: neptunecontrib/viz/__init__.py (hunk 7 above)
Comment (jakubczakon) [reply neptunecontrib/viz/__init__.py]: Done

File: neptunecontrib/versioning/__init__.py (hunk 6 above)
Comment (jakubczakon) [reply neptunecontrib/versioning/__init__.py]: Done

File: neptunecontrib/monitoring/xgboost_monitor.py (hunk 5 above)
Comment (jakubczakon) [reply neptunecontrib/monitoring/xgboost_monitor.py]: Done

File: neptunecontrib/logging/chart.py (hunk 4 above)
Comment (jakubczakon) [reply neptunecontrib/logging/chart.py]: Done

File: neptunecontrib/monitoring/utils.py (hunk 3 above)
Comment (jakubczakon) [reply neptunecontrib/monitoring/utils.py]: Done

File: neptunecontrib/monitoring/utils.py (hunk 2 above)
Comment (jakubczakon) [reply neptunecontrib/monitoring/utils.py]: Done

File: neptunecontrib/api/chart.py (hunk 1 above)
Comment (jakubczakon) [reply neptunecontrib/api/chart.py]: Done

File: neptunecontrib/api/__init__.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (jakubczakon) [reply neptunecontrib/api/__init__.py]: Added __all__ but I am still keeping those `log_chart` `log_table` in the __init__

File: neptunecontrib/api/chart.py [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+        _exp.log_artifact(export_plotly_figure(chart), "charts/" + name + '.html')
+
+    elif is_bokeh_figure(chart):
+        _exp.log_artifact(export_bokeh_figure(chart), "charts/" + name + '.html')
+
+    elif is_altair_chart(chart):
+        _exp.log_artifact(export_altair_chart(chart), "charts/" + name + '.html')
+
+    else:
+        raise ValueError("Currently supported are matplotlib and plotly figures")
```
[the remark this answers is not in the document]
Comment (jakubczakon) [reply neptunecontrib/api/chart.py:140]: done

[the remark this answers is not in the document]
Comment (pitercl) [reply neptunecontrib/api/chart.py:140]: Maybe uncomitted?

**Pull request merged**